src/geometry/euclidean/congruence.lean

```diff
@@ -0,0 +1,303 @@
+/-
+Copyright (c) 2023 . All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jovan Gerbscheid
+-/
+
+import topology.metric_space.isometry
+import geometry.euclidean.triangle
+import linear_algebra.affine_space.finite_dimensional
+
+import tactic.swap_var
+
+/-!
+# Congruences
+
+We define congruence, i.e., the equivalence between sets of points in a metric space where
+all corresponding pairwise distances are the same. The motivating example is triangles in the plane.
+
+## Main results
+
+In the general case we show an `isometry_equiv` between the points: set.range v₁ ≃ᵢ set.range v₂.
+
+In any real affine space, we show that congruence follows from SSS, SAS, and AAS congruence.
+
+To show SSS and SAS, we expand the definition of an angle to write the cosine of it in terms of
+the triangle side lengths (which is basically the same as using the cosine rule).
+Then we show the sine rule, by epressing sin (∠ a b c) as sqrt (1 - cos (∠ a b c)^2), and
+expanding the terms.
+Then we use the sine rule to show AAS congruence.
+
+## Notation
+
+- (![a,b,c] : fin 3) → P represents triangle a b c.
+- (v₁ ≅ v₂ : Prop) represents that (v₁ : ι → P₁) and (v₂ : ι → P₂) are congruent.
+
+-/
+
+noncomputable theory
+open_locale euclidean_geometry
+
+/-- Congruence between indexed sets of vertices v₁ and v₂. Use
+`open_locale congruence` to access the `∠ p1 p2 p3`
+notation. -/
```

Comment:
This doc string has the wrong notation in it (`∠` instead of `≅`). And right now it's referencing the wrong locale, since you're using the `euclidean_geometry` locale, but I think the locale should change anyway (see my next comment).